File: viralt-app/src/types/submission.ts
/** Submission Types -- Types for the contest submission system */

/** Main submission information */
export interface SubmissionInfo {
  submissionId: number;
  campaignId: number;
  clientId: number;
  fileUrl: string;
  fileType: number;
  caption: string | null;
  status: number;
  voteCount: number;
  judgeScore: number | null;
  submittedAt: string;
}

/** Data required to create a new submission (no ID) */
export interface SubmissionInsertInfo {
  campaignId: number;
  clientId: number;
  fileUrl: string;
  fileType: number;
  caption: string | null;
}

/** Submission list API result */
export interface SubmissionListResult {
  submissions: SubmissionInfo[];
  sucesso: boolean;
  mensagem: string | null;
  erros: string[] | null;
}

/** Submission get API result */
export interface SubmissionGetResult {
  submission: SubmissionInfo | null;
  sucesso: boolean;
  mensagem: string | null;
  erros: string[] | null;
}

/** Status-only result */
export interface SubmissionStatusResult {
  sucesso: boolean;
  mensagem: string;
  erros: string[] | null;
}

/** Submission status enum values */
export enum SubmissionStatus {
  Pending = 0,
  Approved = 1,
  Rejected = 2,
}

/** Submission file type enum values */
export enum SubmissionFileType {
  Image = 0,
  Video = 1,
}

File: Viralt.Domain/Models/Submission.cs
namespace Viralt.Domain.Models;

public class Submission
{
    public long SubmissionId { get; set; }
    public long CampaignId { get; set; }
    public long ClientId { get; set; }
    public string FileUrl { get; set; }
    public int FileType { get; set; }
    public string Caption { get; set; }
    public int Status { get; set; }
    public int VoteCount { get; set; }
    public decimal? JudgeScore { get; set; }
    public DateTime SubmittedAt { get; set; }

    public virtual Campaign Campaign { get; set; }
    public virtual Client Client { get; set; }
    public virtual ICollection<Vote> Votes { get; set; } = new List<Vote>();
}
